src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/Dialog/CSharpInterfacePullerViaDialog.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp.Dialog
+{
+    [ExportLanguageService(typeof(AbstractInterfacePullerWithDialog), LanguageNames.CSharp), Shared]
+    internal class CSharpInterfacePullerViaDialog : AbstractInterfacePullerWithDialog
+    {
+        protected override void ChangeMemberToPublicAndNonStatic(DocumentEditor editor, ISymbol symbol, SyntaxNode memberSyntax, SyntaxNode containingTypeNode, ICodeGenerationService codeGenerationService)
+        {
+            var modifier = DeclarationModifiers.From(symbol).WithIsStatic(false);
+            if (symbol is IEventSymbol eventSymbol)
+            {
+                if (memberSyntax.Parent != null &&
+                    memberSyntax.Parent.Parent is BaseFieldDeclarationSyntax eventDeclaration)
+                {
+                    if (eventDeclaration.Declaration.Variables.Count > 1)
+                    {
+                        var options = new CodeGenerationOptions(generateMethodBodies: false, generateMembers: false);
+                        var publicAndNonStaticSymbol = CodeGenerationSymbolFactory.CreateEventSymbol(
```

Comment:
so we have a pattern you should use here.  specifically: https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationSymbolFactory.cs#L419-L427

The idea is that you should pass in the symbol you want to clone, and then only pass the changes you want to make.  this way you don't have to extract and pass along all those `eventSymbol.Member` args.  This is much clearer to read as well.  It's simple to see that w're just making an event the same as an existing event, but with just a couple of things changed.

Reply:
Got you